04_TS_decorator/classes/decorator.ts

```diff
@@ -0,0 +1,38 @@
+import { Student } from "../types/types";
+import { Univercity } from "./building";
+
+interface Commander {
+    execute(action: string, ...args: unknown[]): void;
+}
+
+export class UniversityCommander implements Commander {
+    
+    private univercity: Univercity;
+
+    constructor(univercity: Univercity) {
+        this.univercity = univercity;
+    }
+
+    public execute(action : keyof { getStudentsList: string, getYoungestStudentAge: string, assignStudents: string, graduateStudents: string }, ...args: Student[]): void {
```

Comment:
Please use DO NOT create any additional types for `action` parameter typing: only `University` interface.